oxidation/bindings/generator/templates/binding_electron_meths.rs.j2

```diff
@@ -0,0 +1,437 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 (eventually AGPL-3.0) 2016-present Scille SAS
+{# Macros -#}
+
+{#-
+ # Types conversions macros
+ #}
+
+
+{%- macro ts_type(type) -%}
+{%- if type.kind == "ref" -%}
+{{ ts_type(type.elem) }}
+{%- elif type.kind == "list" -%}
+JsArray
+{%- elif type.kind == "optional" -%}
+JsValue
+{%- elif type.kind == "bool" -%}
+JsBoolean
+{%- elif type.kind == "int" -%}
+JsNumber
+{%- elif type.kind == "float" -%}
+JsNumber
+{%- elif type.kind in ("str", "str_based") -%}
+JsString
+{%- elif type.kind == "bytes" -%}
+JsTypedArray<u8>
+{%- elif type.kind == "struct" -%}
+JsObject
+{%- elif type.kind == "variant" -%}
+JsObject
+{%- elif type.kind == "result" -%}
+JsObject
+{%- else -%}
+{{ raise("Unsupported type %r" % type) }}
+{%- endif -%}
+{%- endmacro -%}
+
+
+{%- macro maybe_ref(js_val, type) -%}
+{%- if type.kind == "ref" and type.elem.kind == "list" -%}
+{{ js_val }}.as_slice()
+{%- elif type.kind == "optional" and type.elem.kind == "ref" -%}
+{{ js_val }}.as_ref()
+{%- elif type.kind == "ref" -%}
+&{{ js_val }}
+{%- else -%}
+{{ js_val }}
+{%- endif -%}
+{%- endmacro -%}
+
+
+{%- macro render_downcasted_js_to_rs(js_val, type, mut_cx_ref="&mut cx") -%}
+{%- if type.kind == "ref" -%}
+{{ render_downcasted_js_to_rs(js_val, type.elem, mut_cx_ref) }}
+{%- elif type.kind == "list" -%}
+{
+    let size = {{ js_val }}.len({{ mut_cx_ref }});
+    let mut v = Vec::with_capacity(size as usize);
+    for i in 0..size {
+        let js_item = {{ js_val }}.get({{ mut_cx_ref }}, i)?;
+        v.push({{ render_downcasted_js_to_rs("js_item", type.elem, mut_cx_ref) | indent(8) }});
+    }
+    v
+}
+{%- elif type.kind == "optional" -%}
+{
+    if {{ js_val }}.is_a::<JsNull, _>({{ mut_cx_ref }}) {
+        None
+    } else {
+        let {{ js_val }} = {{ js_val }}.downcast_or_throw::<{{ ts_type(type.elem) }}, _>({{ mut_cx_ref }})?;
+        Some(
+            {{ render_downcasted_js_to_rs("js_val", type.elem, mut_cx_ref) | indent(12) }}
+        )
+    }
+}
+{%- elif type.kind == "result" -%}
+{
+    if {{ js_val }}.get::<JsBoolean, _, _>({{ mut_cx_ref }}, "ok")?.value() {
+        let js_val = {{ js_val }}.get::<{{ ts_type(type.ok) }}, _, _>({{ mut_cx_ref }}, "value")?;
+        {{ render_downcasted_js_to_rs("js_val", type.ok, mut_cx_ref) | indent(8) }}
+    } else {
+        let js_val = {{ js_val }}.get::<{{ ts_type(type.err) }}, _, _>({{ mut_cx_ref }}, "error")?;
+        {{ render_downcasted_js_to_rs("js_val", type.err, mut_cx_ref) | indent(8) }}
+    }
+}
+{%- elif type.kind == "int" -%}
+{{ js_val }}.value({{ mut_cx_ref }}) as i64
+{%- elif type.kind == "str" -%}
+{{ js_val }}.value({{ mut_cx_ref }})
+{%- elif type.kind == "str_based" -%}
+match {{ js_val }}.value({{ mut_cx_ref }}).parse() {
+    Ok(val) => val,
+    Err(err) => {
+        ({{ mut_cx_ref }}).throw_type_error(err)?;
+        unreachable!();
```

Comment:
```suggestion
        return Err(({{ mut_cx_ref }}).throw_type_error(err).expect_err("Should have thrown an error"));
```
WDYT ? It's more explicit that `throw_type_error` should always return an `Error`

Reply:
You're right, this is much better ! done.